tsl/src/bgw_policy/drop_chunks_api.c

```diff
@@ -72,6 +75,21 @@ drop_chunks_add_policy(PG_FUNCTION_ARGS)
 				 errhint("Please add the policy to the corresponding uncompressed hypertable "
 						 "instead.")));
 
+	/* Verify that the hypertable owner can create a background worker */
+	role_tup = SearchSysCache1(AUTHOID, ObjectIdGetDatum(owner_id));
+	rform = (Form_pg_authid) GETSTRUCT(role_tup);
+
+	if (!rform->rolcanlogin)
+	{
+		ReleaseSysCache(role_tup);
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_AUTHORIZATION_SPECIFICATION),
+				 errmsg("Unable to start drop chunks background process for hypertabletable \"%s\"",
+						hypertable->fd.table_name.data),
+				 errhint("Hypertable owner must have LOGIN permission to run background tasks")));
+	}
+	ReleaseSysCache(role_tup);
```

Comment:
This is repeated three times with slight variations. Probably a good candidate for a separate function to do the job.

Reply:
Done.